controllers/affiliation_controller.js

```diff
@@ -0,0 +1,76 @@
+const Affiliation = require("../models/affiliation_model.js");
+const Ambassador = require("../models/ambassador_model.js");
+const { SchemaTypes, Schema } = require("mongoose");
+
+module.exports.createAffiliation = async (req, res, next) => {
+  try {
+    const { AmbassadorId } = req.body;
+
+    // Génération du code unique pour le lien d'affiliation
+    const affiliateCode = Math.random().toString(36).substring(2);
+
+    // Ajout du code au champ "lienAffiliation" dans la base de données
+    await Ambassador.findByIdAndUpdate(AmbassadorId, {
+      affiliateCode: affiliateCode,
+    });
+
+    // Construction du lien complet avec le nom de domaine
+    const affiliateLink = `$dev-style.com/affiliate/${affiliateCode}`;
+
+    // return res.status(200).json({ link: affiliateLink });
```

Comment:
Therefore, as I said in the file ```affiliation_model.js```, affiliateLink is redundant, so it won't be useful.
Remove.